### PR TITLE
[ALLUXIO-1878] Support additional HDFS read methods.

### PR DESCRIPTION
**Pull request by jsimsa**
https://alluxio.atlassian.net/browse/ALLUXIO-1878

### CODE REVIEW

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java [hunk 1]
```diff
@@ -178,19 +178,19 @@ public int read() throws IOException {
   }
 
   @Override
-  public int read(byte[] b) throws IOException {
-    throw new IOException(ExceptionMessage.NOT_SUPPORTED.getMessage());
+  public int read(byte[] buffer) throws IOException {
+    return read(buffer, 0, buffer.length);
```
Comment (yupeng9) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: Shall we add tests to cover these methods?

Comment (jsimsa) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: done

**Comment by jsimsa** [comment]
@yupeng9 addressed your comment (and fix incorrect implementation of readFully ... previously, it would not do the right thing for partial reads)

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java [hunk 2]
```diff
@@ -283,30 +283,21 @@ private int readFromHdfsBuffer() throws IOException {
...
-   * @param position N/A
-   * @param buffer N/A
-   * @param offset N/A
-   * @param length N/A
-   * @throws IOException always
-   */
   @Override
   public void readFully(long position, byte[] buffer, int offset, int length) throws IOException {
-    throw new IOException(ExceptionMessage.NOT_SUPPORTED.getMessage());
+    int nread = 0;
```
Comment (gpang) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: NIT: maybe bytesRead? or totalBytesRead?

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java [hunk 3]
```diff
@@ -283,30 +283,21 @@ private int readFromHdfsBuffer() throws IOException {
...
+    while (nread < length) {
+      int ret = read(position + nread, buffer, offset + nread, length - nread);
+      if (ret == -1) {
```
Comment (gpang) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: Is this still allowable if you read the EOF before you finish reading the 'length' amount of data?

**Comment by gpang** [comment]
Thanks for the fix! I left a few minor comments.

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java (hunk 2 above)
Comment (jsimsa) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: done (I used a shorter variable name, but added a comment)

File: core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java (hunk 3 above)
Comment (jsimsa) [inline core/client/src/main/java/alluxio/hadoop/HdfsFileInputStream.java]: good catch, as per https://docs.oracle.com/javase/7/docs/api/java/io/DataInput.html#readFully(byte[]) we need to throw EOFException here

**Comment by jsimsa** [comment]
thanks @gpang, addressed your comments

**Comment by gpang** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**